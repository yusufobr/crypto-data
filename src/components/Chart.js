/* eslint-disable */
import React from 'react';
import {
  AreaChart,
  Area,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  ResponsiveContainer,
} from 'recharts';

function Chart(props) {
  const { history } = props;
  return (
    <ResponsiveContainer width="95%" height={300}>
    <AreaChart
      data={history}
      margin={{
        top: 10,
        right: 30,
        left: 0,
        bottom: 0,
      }}
    >
      <CartesianGrid strokeDasharray="3 3" />
      <XAxis dataKey="Date" />
      <YAxis />
      <Tooltip />
      <Area type="monotone" dataKey="USD" stroke="black" fill="#1dd79c" />
    </AreaChart>
    </ResponsiveContainer>
  );
}

export default Chart;
