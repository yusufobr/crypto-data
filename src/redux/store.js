import { configureStore } from '@reduxjs/toolkit';
import coinsReducer from './coin/coinsSlice';

const store = configureStore({
  reducer: {
    coins: coinsReducer,
  },
});

export default store;
