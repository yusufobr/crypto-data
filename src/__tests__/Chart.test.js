import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Chart from '../components/Chart';

class ResizeObserver {
  observe() {}

  unobserve() {}

  disconnect() {}
}

window.ResizeObserver = ResizeObserver;

describe('Chart.js', () => {
  it('should render correctly', () => {
    const data = [
      { USD: 10, Date: '2023-03-13' },
      { USD: 12, Date: '2023-03-14' },
      { USD: 18, Date: '2023-03-15' },
    ];
    const tree = renderer
      .create(
        <Provider store={store}>
          <Chart history={data} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
