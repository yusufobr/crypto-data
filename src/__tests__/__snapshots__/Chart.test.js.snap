// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart.js should render correctly 1`] = `
<div
  className="recharts-responsive-container"
  style={
    Object {
      "height": 300,
      "maxHeight": undefined,
      "minHeight": undefined,
      "minWidth": 0,
      "width": "95%",
    }
  }
/>
`;
